ansys/dpf/core/collection.py

```diff
@@ -52,65 +66,103 @@ def set_labels(self, labels):
                 ['time','complex']
 
         """
         if len(self._info['labels'])!=0:
-            print("the collection has already labels :",self._info['labels'],'deleting existing abels is  not implemented yet')
+            print("the collection has already labels :",self._info['labels'],'deleting existing labels is  not implemented yet')
             return
         request = collection_pb2.UpdateLabelsRequest()
         request.collection.CopyFrom(self._message)
-        request.labels.labels.extend(labels)
+        request.labels.extend([collection_pb2.NewLabel(label=lab) for lab in labels])
         self._stub.UpdateLabels(request)
 
-    def add_label(self, label):
+    def add_label(self, label, default_value =None):
         """Add the requested label to scope the collection
 
         Parameters
         ----------
-        label (optional) : string
+        label : str
             Labels on which the entries will be scoped, for example ``'time'``.
+            
+        default_value : int , optional
+            default value set for existing fields in the collection
 
         Examples
         --------
+        >>> from ansys.dpf import core as dpf
+        >>> coll = dpf.Collection(dpf.types.field)
         >>> coll.add_label('time')
+        
         """
         request = collection_pb2.UpdateLabelsRequest()
         request.collection.CopyFrom(self._message)
-        request.labels.labels.extend([label])
+        new_label = collection_pb2.NewLabel(label=label)
+        if default_value is not None:
+            new_label.default_value.default_value=default_value
+        request.labels.extend([new_label])
         self._stub.UpdateLabels(request)
 
     def _get_labels(self):
         """get the labels scoping the collection
 
         Returns
         -------
-        labels: list(string)
+        labels: list[string]
             labels on which the entries are scoped, for example:
                 ``['time', 'complex']``
         """
         return self._info['labels']
 
     labels = property(_get_labels, set_labels, "labels")
+    
+    def has_label(self, label):
+        """Check if a collection has a specified label
+
+        Parameters
+        ----------
+        label: string
+            Labels that must be searched, for example ``'time'``.
+            
+        Returns
+        -------
+        bool: True if the specified value have been found in the collection. 
+
+        Examples
+        --------
+        >>> from ansys.dpf import core as dpf
+        >>> coll = dpf.Collection(dpf.types.field)
+        >>> coll.add_label('time')
+        >>> coll.has_label('time')
+        True
+        >>> coll.has_label('complex')
+        False
+        
+        """
+        result = False
+        labels = self.labels
+        i = 0
+        while i < len(labels):
+            if labels[i] == label:
+                result = True
+            i += 1
+        return result
```

Comment:
```suggestion
        return label in self.labels
```
https://stackoverflow.com/questions/7571635/fastest-way-to-check-if-a-value-exists-in-a-list